src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -218,11 +219,11 @@
                              "environment" {:type :string
                                             :queryable? true
                                             :field :env.environment}
-                             "value_integer" {:type :number
+                             "value_integer" {:type :integer
                                               :query-only? true
                                               :queryable? false
                                               :field :fv.value_integer}
-                             "value_float" {:type :number
+                             "value_float" {:type :float
```

Comment:
We handle `floats` and `integers` differently in array construction, so we need to differentiate now.